pkg/apis/eventing/v1alpha1/trigger_validation.go

```diff
@@ -39,9 +41,28 @@ func (ts *TriggerSpec) Validate(ctx context.Context) *apis.FieldError {
 		errs = errs.Also(fe)
 	}
 
-	if ts.Filter != nil && ts.Filter.SourceAndType == nil {
-		fe := apis.ErrMissingField("filter.sourceAndType")
-		errs = errs.Also(fe)
+	if ts.Filter != nil {
+		filtersSpecified := make([]string, 0)
+
+		if ts.Filter.DeprecatedSourceAndType != nil {
+			filtersSpecified = append(filtersSpecified, "filter.sourceAndType")
+		}
+
+		if ts.Filter.Attributes != nil {
+			filtersSpecified = append(filtersSpecified, "filter.attributes")
+			if len(*ts.Filter.Attributes) == 0 {
+				fe := &apis.FieldError{
+					Message: "At least one filtered attribute must be specified",
+					Paths:   []string{"filter.attributes"},
+				}
+				errs = errs.Also(fe)
+			}
+		}
```

Comment:
Validate the name of the attributes match `[a-z][0-9a-z]*`, which I think is the MUST portion of the [0.3 CloudEvents spec](https://github.com/cloudevents/spec/blob/v0.3/spec.md#attribute-naming-convention).

Otherwise it will be really easy for someone to create a Trigger filtering on the attribute `Foo`, which will never match because the CloudEvents SDK will normalize it to `foo`.

Reply:
Done, checking the size as well